core/commands/add.go

```diff
@@ -277,16 +318,24 @@ func addDir(n *core.IpfsNode, dir files.File, out chan interface{}, progress boo
 			break
 		}
 
-		node, err := addFile(n, file, out, progress, false)
-		if err != nil {
+		node, err := addFile(n, file, out, progress, false, hidden, localIgnorePatterns)
+		if _, ok := err.(*hiddenFileError); ok {
+			// hidden file error, set the node to nil for below
+			node = nil
+		} else if _, ok := err.(*ignoreFileError); ok {
+			// ignore file error, set the node to nil for below
+			node = nil
+		} else if err != nil {
```

Comment:
so this just silently skips it? so would:

```
echo foo >.ipfsignore
ipfs add foo
```
just silently succeed without doing anything?

on this very specific case, git gives an error like:

```
echo foo >.gitignore
> git add foo
The following paths are ignored by one of your .gitignore files:
foo
Use -f if you really want to add them.
fatal: no files added
```